[CI:DOCS] options/network: fix markdown lists

baltitenger opened the pull request:
<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]". That will prevent functional tests from running and save time and energy.

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
None
```

Luap99 commented on docs/source/markdown/options/network.md:
```diff
@@ -15,17 +15,23 @@ Valid _mode_ values are:
-  For example, to set a static ipv4 address and a static mac address, use `--network bridge:ip=10.88.0.10,mac=44:33:22:11:00:99`.
-
-- \<network name or ID\>[:OPTIONS,...]: Connect to a user-defined network; this is the network name or ID from a network created by **[podman network create](podman-network-create.1.md)**. Using the network name implies the bridge network mode. It is possible to specify the same options described under the bridge mode above. Use the **--network** option multiple times to specify additional networ ...
+    For example, to set a static ipv4 address and a static mac address, use `--network bridge:ip=10.88.0.10,mac=44:33:22:11:00:99`.
```
I don't think this looks right, it intends this to the same level as the interface_name but this clearly should be on the level from bridge which it is right now.

baltitenger replied:
Yea, it looks wrong in the text but the generated output is correct. Markdown takes 4 spaces as a unit of indentation.

Luap99 replied:
What generated output, html or man page?
It should not be on the nested list level as it applies to the bridge not interface_name so I think this change is incorrect

baltitenger replied:
I'm looking at the generated manpage and there it's correct, the example is on the same level as `bridge[...]`, not `interface_name`. However, in the html it's indeed incorrect. Not sure if this can be resolved.

Luap99 replied:
yeah I see it now, I was only looking at the markdown/html view.

Luap99 commented on docs/source/markdown/options/network.md:
```diff
@@ -15,17 +15,23 @@ Valid _mode_ values are:
-- \<network name or ID\>[:OPTIONS,...]: Connect to a user-defined network; this is the network name or ID from a network created by **[podman network create](podman-network-create.1.md)**. Using the network name implies the bridge network mode. It is possible to specify the same options described under the bridge mode above. Use the **--network** option multiple times to specify additional networ ...
+    For example, to set a static ipv4 address and a static mac address, use `--network bridge:ip=10.88.0.10,mac=44:33:22:11:00:99`.
 
+- **\<network name or ID\>[:OPTIONS,...]**: Connect to a user-defined network; this is the network name or ID from a network created by **[podman network create](podman-network-create.1.md)**. Using the network name implies the bridge network mode. It is possible to specify the same options described under the bridge mode above. Use the **--network** option multiple times to specify additional ne ...
   For backwards compatibility it is also possible to specify comma-separated networks on the first **--network** argument, however this prevents you from using the options described under the bridge section above.
```
Doesn't this just remove the new paragraph? IMO the extra paragraph makes it easier to read.

baltitenger replied:
Well, I couldn't get new paragraphs to work the way they should. If it's a separate paragraph, it seems to indent it way less than the list itself. Lists seem to be buggy anyway with gomd2man, they are all indented a step unnecessarily.

baltitenger replied:
I agree that it would be nicer to replace all forced newlines (`\` at eol) with new paragraphs but I don't think there's a way to do it right now, `gomd2man` seems to merge them into a single paragraph for some reason. (btw, couldn't sphinx generate man pages as well? not that it would necessarily solve this)

baltitenger replied:
Here's why: https://github.com/cpuguy83/go-md2man/blob/master/md2man/roff.go#L144-L147

I'm not sure if that comment is correct though.

Luap99 replied:
I am fine with this change if it breaks man pages. I have slight preference for the paragraph but if the generation cannot handle it today then it is fine.

baltitenger replied:
Apparently it's a pretty simple fix, check out https://github.com/baltitenger/podman/tree/man-network-fix-2. There I also patched the man page generator to handle paragraphs in list items. I'm not sure if it would have an impact anywhere else, would be best to check the diff before/after. I guess this change should be separated from the docs changes as well.

baltitenger replied:
I have submitted a pr to cpuguy83/go-md2man#116. Not sure how active that project is, so I guess in the meantime this pr could be merged as is and updated once go-md2man has a release with this merged. Or you can patch the module, idk.

Luap99 replied:
I rather not use a patched go-md2man here, agreed for now your changes are good, thanks!

Luap99 reviewed (commented):
Can you describe what you are actually "fixing"?

baltitenger commented:
I updated the commit message and also changed parts of options to italic so it's clearer what part of them needs to be changed and what's typed verbatim as per man(1).

Btw the `[CI:DOCS]` prefix doesn't seem to do its thing, it runs all the tests anyway.

baude commented:
> Btw the `[CI:DOCS]` prefix doesn't seem to do its thing, it runs all the tests anyway.

It most certainly did not ...

baude commented on docs/source/markdown/options/network.md:
```diff
@@ -59,6 +65,7 @@ Valid _mode_ values are:
     **-U none** are given to disable the same functionality from container to
     host. \
     Some examples:
+
```
if you repush, im not sure this needs to be here?

baltitenger replied:
Yep, it seems to render fine without it.

baude commented:
small NIT, LGTM otherwise.

Luap99 commented on docs/source/markdown/options/network.md:
```diff
@@ -9,32 +9,38 @@ Set the network mode for the <<container|pod>>.
+  - **ip=**_IPv4_: Specify a static ipv4 address for this container.
+  - **ip=**_IPv6_: Specify a static ipv6 address for this container.
+  - **mac=**_MAC_: Specify a static mac address for this container.
+  - _interface\_name_: Specify a name for the created network interface inside the container.
```
pre-existing but this is incorrect, it should be `**interface_name=**_name_:`

baltitenger replied:
Fixed in the next push.

Luap99 reviewed (approved):
/lgtm
